### ISSUE TITLE
"Cannot decode introspection" error after updating to 3.7.5

### ISSUE THREAD

**Issue by ikurek** [apollographql/apollo-kotlin#4859]
### Version

3.7.5 and above

### Summary

After upgrading from `3.7.4` to `3.7.5` Apollo fails to download introspection schema. The `--stacktrace` and `--debug` options do not provide any valuable information besides two error messages: `Invalid introspection schema` and `Cannot decode introspection`. The same `downloadDevApolloSchemaFromIntrospection` task that works without issues in `3.7.4` fails on both my machine and Bitrise instances. Here's the introspection configuration we are using:

```kotlin
apollo {
    service("dev") {
        generateKotlinModels.set(true)
        generateAsInternal.set(true)
        packageName.set("com.my.package.core.network")
        schemaFile.set(File("$projectDir/src/main/graphql/schema.json"))
        introspection { endpointUrl.set(graphQlUrl) }
        outputDirConnection { connectToKotlinSourceSet("dev") }
    }
}
```

Looking at https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.5 there were some changes related to schema downloads and introspection, it may be possible one of them introduced some sort of regression?

### Steps to reproduce the behavior

Rollback to version `3.7.4` resolves the issue entirely, using the same schema URL and apollo configuration. The schema contains some deprecated fields, which may be a potential issue, since version `3.7.5` introduced changes to handling deprecated fields. Sadly, I can't share the entire schema file.

### Logs

```
> Task :core-network:downloadDevApolloSchemaFromIntrospection FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core-network:downloadDevApolloSchemaFromIntrospection'.
> Cannot decode introspection 

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':core-network:downloadDevApolloSchemaFromIntrospection'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
        ...
Caused by: java.lang.RuntimeException: Cannot decode introspection 
        at com.apollographql.apollo3.relocated.com.apollographql.apollo3.ast.introspection.IntrospectionSchemaKt.toIntrospectionSchema(SourceFile:68)
        at com.apollographql.apollo3.gradle.internal.ApolloDownloadSchemaTask.taskAction(SourceFile:689)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
        ...
Caused by: java.lang.IllegalArgumentException: Invalid introspection schema: 
        at com.apollographql.apollo3.relocated.com.apollographql.apollo3.ast.introspection.IntrospectionSchemaKt.toIntrospectionSchema(SourceFile:66)
        ...

**Comment by martinbonnin** [comment]
Thanks for sending this. My hunch is that the Apollo Plugin uses unsupported introspection types, your server returns `{ "data": null }` and this triggers the error. We should definitely check for errors. I'll look into that

**Comment by martinbonnin** [comment]
@ikurek can you try with the [SNAPSHOTs](https://github.com/apollographql/apollo-kotlin#snapshots)? [3.8.1-SNAPSHOT](https://s01.oss.sonatype.org/content/repositories/snapshots/com/apollographql/apollo3/apollo-runtime/3.8.1-SNAPSHOT/) should be available with a fix

**Comment by vrrashkov** [comment]
@martinbonnin  Just had the same issue happen to me and was happy to see that it was already addressed. I can confirm that when I tested with the **3.8.1-SNAPSHOT** version it works on my end and the 3.8.0 doesn't. Thanks!

**Comment by BoD** [comment]
The fix is now released as part of v3.8.1
